spec/ruby/core/module/refine_spec.rb

```diff
@@ -760,7 +785,140 @@ def foo
         result = refined_class.new.foo
       end
 
-      result.should == "foo"
+      result.should == [:R2, :C]
+    end
+
+    it "looks only in the refined class even if there is another active refinement" do
+      refined_class = ModuleSpecs.build_refined_class(for_super: true)
+
+      refinement = Module.new do
+        refine refined_class do
+          def bar
+            "you cannot see me from super because I belongs to another active R"
+          end
+        end
+      end
+
+      refinement_with_super = Module.new do
+        refine refined_class do
+          def bar
+            super
+          end
+        end
+      end
+
+      -> {
+        Module.new do
+          using refinement
+          using refinement_with_super
+          refined_class.new.bar
+        end
+      }.should raise_error(NoMethodError)
```

Comment:
`should raise_error` should be used on the narrowest scope possible, so here around `instance.bar`